src/components/core/admin/collectFeesHandler.ts

```diff
@@ -0,0 +1,153 @@
+import { AdminHandler } from './adminHandler.js'
+import { AdminCollectFeesCommand } from '../../../@types/commands.js'
+import { P2PCommandResponse } from '../../../@types/OceanNode.js'
+import {
+  ValidateParams,
+  buildInvalidParametersResponse,
+  buildErrorResponse,
+  buildInvalidRequestMessage,
+  validateCommandParameters
+} from '../../httpRoutes/validateCommands.js'
+import {
+  getConfiguration,
+  checkSupportedChainId,
+  Blockchain
+} from '../../../utils/index.js'
+import { getProviderWallet } from '../utils/feesHandler.js'
+import { parseUnits, Contract, ZeroAddress, isAddress } from 'ethers'
+import ERC20Template from '@oceanprotocol/contracts/artifacts/contracts/templates/ERC20Template.sol/ERC20Template.json' assert { type: 'json' }
+import { CORE_LOGGER } from '../../../utils/logging/common.js'
+import { Readable } from 'stream'
+
+export class CollectFeesHandler extends AdminHandler {
+  validate(command: AdminCollectFeesCommand): ValidateParams {
+    if (
+      !validateCommandParameters(command, [
+        'chainId',
+        'tokenAddress',
+        'tokenAmount',
+        'destinationAddress'
+      ])
+    ) {
+      return buildInvalidRequestMessage(
+        `Missing chainId field for command: "${command}".`
+      )
+    }
+    if (!isAddress(command.tokenAddress) || !isAddress(command.destinationAddress)) {
+      const msg: string = `Invalid format for token address or destination address.`
+      CORE_LOGGER.error(msg)
+      return buildInvalidRequestMessage(msg)
+    }
+    return super.validate(command)
+  }
+
+  async handle(task: AdminCollectFeesCommand): Promise<P2PCommandResponse> {
+    const validation = this.validate(task)
+    if (!validation.valid) {
+      return buildInvalidParametersResponse(validation)
+    }
+    const config = await getConfiguration()
+    if (task.node && task.node !== config.keys.peerId.toString()) {
+      const msg: string = `Cannot run this command ${JSON.stringify(
+        task
+      )} on a different node.`
+      CORE_LOGGER.error(msg)
+      return buildErrorResponse(msg)
+    }
+    const checkChainId = await checkSupportedChainId(task.chainId)
+    if (!checkChainId.validation) {
+      return buildErrorResponse(
+        `Chain ID ${task.chainId} is not supported in the node's config`
+      )
+    }
+
+    try {
+      const { rpc, network, chainId, fallbackRPCs } =
+        config.supportedNetworks[task.chainId]
+      const blockchain = new Blockchain(rpc, network, chainId, fallbackRPCs)
+      const provider = blockchain.getProvider()
+      const providerWallet = await getProviderWallet(String(task.chainId))
+      const ammountInEther = parseUnits(task.tokenAmount.toString(), 'ether')
+      const providerWalletAddress = await providerWallet.getAddress()
+
+      let receipt
+      if (task.tokenAddress === ZeroAddress) {
```

Comment:
Can you make sure that all the address comparisons are done in lowercase? 

```suggestion
      if (task.tokenAddress?.toLowerCase() === ZeroAddress) {
```

Reply:
tokenAddress is a mandatory parameter and validated that it's not missing, I'll add `toLowerCase()`, thanks.